app/serializers/serialize_exercise_instructions.rb

```diff
@@ -0,0 +1,72 @@
+class SerializeExerciseInstructions
+  include Mandate
+
+  initialize_with :exercise
+
+  def call
+    {
+      overview: overview,
+      general_hints: general_hints,
+      tasks: tasks
+    }
+  end
+
+  private
+  def overview
+    instructions_doc.each.
+      take_while { |node| node.type != :header }.
+      map(&:to_html).
+      join
+  end
+
+  def general_hints
+    hints["general"].to_a
+  end
+
+  memoize
+  def hints
+    hints_doc.each_cons(2).each_with_object({}) do |(header, list), hints|
+      next unless header.type == :header
+      next unless header.header_level == 2
+      next unless list.type == :list
+
+      heading = parse_title(header).downcase
+      hints[heading] = list.each.map { |list_item| list_item.each.first.to_html }
+    end
+  end
+
+  def tasks
+    instructions_doc.each.
+      drop_while { |node| node.type != :header }.
+      chunk_while { |_, nxt| nxt.type != :header }.
+      each_with_object([]) do |nodes, tasks|
+        task_title = parse_title(nodes.first)
+        task_text = nodes.drop(1).each.map(&:to_html).join.strip
```

Comment:
Better to not mutate the array. I can't remember if it's `1..-1`, or `1, -1`
```suggestion
        task_text = nodes[1,-1].each.map(&:to_html).join.strip
```

Reply:
Oh, that mutates? I thought that a mutating method would have had a `!` postfix. I'll fix it

Reply:
`[1..]` is the most concise version (suggested by Rubocop over `[1..-1]`.

Reply:
That's nice. Didn't know of that.

Reply:
`.drop` does not mutate, but `[1..]` is something I see much more in the wild, so probably what you want here.